go/consensus/tendermint/apps/roothash/roothash.go

```diff
@@ -635,23 +661,7 @@ func (app *rootHashApplication) tryFinalizeExecutorCommits(
 
 		return nil
 	case commitment.ErrDiscrepancyDetected:
-		// Discrepancy has been detected.
-		ctx.Logger().Warn("executor discrepancy detected",
-			"round", round,
-			logging.LogEvent, roothash.LogEventExecutionDiscrepancyDetected,
-		)
-
-		tagV := ValueExecutionDiscrepancyDetected{
-			ID: runtime.ID,
-			Event: roothash.ExecutionDiscrepancyDetectedEvent{
-				Timeout: forced,
-			},
-		}
-		ctx.EmitEvent(
-			tmapi.NewEventBuilder(app.Name()).
-				Attribute(KeyExecutionDiscrepancyDetected, cbor.Marshal(tagV)).
-				Attribute(KeyRuntimeID, ValueRuntimeID(runtime.ID)),
-		)
+		// This was already handled above, so it should not happen.
```

Comment:
control shouldn't reach here, or just nothing left to do? we don't return early in the `if` added above

Reply:
So the branch added above will overwrite `err` and the repeated `TryFinalize` cannot (currently) return the "discrepancy detected" error.